pkg/ottl/contexts/ottldatapoints/datapoints.go

```diff
@@ -233,8 +234,7 @@ func accessMetricType() ottl.StandardGetSetter {
 			return int64(ctx.(TransformContext).GetMetric().DataType())
 		},
 		Setter: func(ctx ottl.TransformContext, val interface{}) {
-			// TODO Implement methods so correctly convert data types.
-			// https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/10130
+			fmt.Errorf("setting metric type is not allowed")
```

Comment:
This should be a call to `fmt.Println`, `Errorf` just creates an error from the string.